File: LAB 0/lab0.java
import java.io.*;
import java.util.InputMismatchException;
import java.util.Random;
import java.util.Scanner;

public class Main {

    public static void generate_file(String fileName, int size, int min, int max) throws IOException {
        try {
            File file = new File(fileName);
            if (file.createNewFile()) {
                System.out.println("Fisierul " + file.getName() + " creat cu succes");
                Random random  = new Random();
                FileWriter writer = new FileWriter(fileName);
                    for (int j = 0; j < size; j++) {
                        int x = random.nextInt(max - min) + min;
                        writer.write(String.valueOf(x) + '\n');
                    }
                writer.close();
            }
            else {
                System.out.println("Fisierul exista deja");
            }
        }
        catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static boolean verify_equal(String fileName1, String fileName2) throws IOException {
        FileReader reader1 = new FileReader(new File(fileName1));
        FileReader reader2 = new FileReader(new File(fileName2));
        BufferedReader buffer1 = new BufferedReader(reader1);
        BufferedReader buffer2 = new BufferedReader(reader2);
        try {
            String data1 = buffer1.readLine();
            String data2 = buffer2.readLine();
            while (data1 != null && data2 != null) {
                if (!data1.equals(data2))
                    return false;
                data1 = buffer1.readLine();
                data2 = buffer2.readLine();
                System.out.println(data1 + " " + data2);
            }
            if (data1 != data2) {
                return false;
            }
        }
        catch(Exception e){
            return false;
        }
        return true;
    }


    public static void main(String[] args) throws IOException {
        while (true){
            Scanner scanner = new Scanner(System.in);
            System.out.print("\n1. Genereaza fisier\n2. Verifica similaritatea a doua fisiere\n3. Iesire din meniu\nOptiune : ");
            try {
                int selection = scanner.nextInt();
                switch (selection) {
                    case 1 :
                        System.out.print("Nume fisier : ");
                        String fileName = scanner.next();
                        System.out.print("Dimensiune :");
                        int size = scanner.nextInt();
                        System.out.print("Minim :");
                        int min = scanner.nextInt();
                        System.out.print("Maxim :");
                        int max = scanner.nextInt();
                        Main.generate_file(fileName, size, min, max);
                        break;
                    case 2:
                        System.out.print("Nume fisier 1: ");
                        String fileName1 = scanner.next();
                        System.out.print("Nume fisier 2: ");
                        String fileName2 = scanner.next();
                        Main.verify_equal(fileName1, fileName2);
                        break;
                    case 3:
                        return;
                    default:
                        System.out.println("Optiune invalida!");
                        break;
                }
            }
            catch (InputMismatchException e) {
                System.out.println("Optiune invalida!");
            }

        }
    }
}

File: LAB 0/lab0.cpp
#include <iostream>
#include <string>
#include <random>
#include <fstream>
using std::string;
using std::ifstream;
using std::ofstream;
using std::mt19937;
using std::uniform_int_distribution;
using std::cout;
using std::cin;
using std::exception;
using std::ostream;
using std::endl;


void generate_file(string filename, int size, int min, int max) {
    ofstream out(filename);
    mt19937 mt{ std::random_device{}() };
    uniform_int_distribution<> dist(min, max);

    for (int i = 0; i < size; i++) {
        int rndNr = dist(mt);
        out << rndNr << "\n";
    }
    out.close();
}


bool equal(string filename1, string filename2) {
    ifstream in1(filename1);
    ifstream in2(filename2);
    string data1, data2;
    while ((in1 >> data1) && (in2 >> data2)) {
        if (data1 != data2) {
            return false;
        }
    }
    if (!in1.eof() || !in2.eof()) {
        return false;
    }
    return true;
}


int main()
{
	int cmd = 1;
	while (true) {
		cout << "\n1. Generare fisier\n2. Compararea a doua fisiere\n3. Iesire\n\nComanda : ";
		cin >> cmd;
		try {
			switch (cmd) {

			case 1:
			{
				string filename;
				int size, min, max;
				cout << "Nume fisier : ";
				cin >> filename;
				cout << "Dimensiune : ";
				cin >> size;
				cout << "Minim : ";
				cin >> min;
				cout << "Maxim : ";
				cin >> max;
				generate_file(filename, size, min, max);
				break;
			}

			case 2:
			{
				string filename1, filename2;
				cout << "Nume fisier 1: ";
				cin >> filename1;
				cout << "Nume fisier 2: ";
				cin >> filename2;
				if (equal(filename1, filename2)) {
					cout << "Fisierele au acelasi continut!";
				}
				else {
					cout << "Fisierele sunt diferite!";
				}
				break;
			}

			case 3:
			{
				return 0;
			}
				
			default:
			{
				cout << "Optiune invalida!";
				break;
			}
				
			}
		}
		catch (exception& ex) {
			cout << ex.what();
			cout << endl;
		}
	}
	return 0;
}
